burn-common/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod id;
+pub mod rand;
+pub mod stub;
+
+extern crate alloc;
```

Comment:
Do we need the alloc crate when std is activated? Does it change anything?

Reply:
No, it does not change anything. Rust reexports core and alloc to std. no_std hides std name space. 

alloc crate is needed because the code references Vec and String from alloc. 

BTW, this is how NDArray does it: https://github.com/rust-ndarray/ndarray/blob/master/src/lib.rs#L115

(but I don't like that they re-export std as core because it's a lie and compilation errors look weird). 

This is also a recommended way in this book: https://www.lurklurk.org/effective-rust/no-std.html